### PR TITLE
Feat: Audio Controller added with registerAudio, play and stop function

### PR DESCRIPTION
**Pull request by rituraj-wednesday**
Feat: Audio Controller added with registerAudio, play and stop function

- registerAudio with response based throttling
- Partial content (206) handled using AudioContext
- play and stop song on click

Audio State Handling:
PR4-audio-controller

### CODE REVIEW

**Comment by rituraj-wednesday** [comment]
New UI changes added:

- Play/Pause function updated
- Title for the track
- Icon Button shouldn't collapse while playing

UI changes:
PR4-ui-changes

File: app/components/Track/index.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+`;
+
+const ButtonFlex = styled.div`
+  display: flex;
+  height: 100%;
+`;
+
+const TitleWrapper = styled.div`
+  color: white;
+  font-size: 24px;
```
Comment (rajatdangat-ws) [inline app/components/Track/index.js]: Please use rem as unit for font size so that it is responsive

File: app/components/Track/index.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                margin: 'calc(30% - 4px) 8px auto auto',
+                height: '60px',
+                width: '60px'
+              }}
+              aria-label={`${isPlaying ? translate('pauseText') : translate('playText')} ${translate('buttonText')}`}
+              onClick={onPlayPauseClick}
+            >
+              <If
+                condition={!isPlaying}
+                otherwise={<PauseCircleFilledRoundedIcon sx={{ color: 'white', fontSize: '60px' }} />}
```
Comment (rajatdangat-ws) [inline app/components/Track/index.js]: use rem for font size

File: app/components/Track/index.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+              >
+                <PlayCircleFilledRoundedIcon sx={{ color: 'white', fontSize: '60px' }} />
```
Comment (rajatdangat-ws) [inline app/components/Track/index.js]: use rem for font size

File: app/components/Track/index.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+          </InfoIconWrapper>
+          <InfoIconWrapper>
+            <IconButton
+              sx={{
+                margin: 'calc(30% - 4px) auto auto 8px',
+                height: '60px',
+                width: '60px'
+              }}
+            >
+              <InfoRoundedIcon sx={{ color: 'white', fontSize: '60px' }} />
```
Comment (rajatdangat-ws) [inline app/components/Track/index.js]: use rem for font size

File: app/components/Track/tests/index.test.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  it('should contain 1 Track component', () => {
+    const dummyTrack = { trackName: 'hello', artworkUrl100: 'artWork' };
+    const { getAllByLabelText } = renderWithIntl(<Track  track={dummyTrack}/>);
+    expect(getAllByLabelText(dummyTrack.trackName).length).toBe(1);
+  });
+
+  it('should call play audio on Play Button click', () => {
+    const element = renderWithIntl(<Track track={{ trackName: 'hello', artworkUrl100: 'artWork', trackId: 1 }} currentTrackID={2} />);
+    const playButtonEle = element.getAllByLabelText(`${translate('playText')} ${translate('buttonText')}`);
+    console.log(playButtonEle.length);
```
Comment (rajatdangat-ws) [inline app/components/Track/tests/index.test.js]: remove the log if not required

File: app/components/Track/tests/index.test.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    expect(playButtonEle.length).toBe(1);
+    fireEvent.click(playButtonEle[0]);
+    expect(mockedPlay).toHaveBeenCalledWith({ trackName: 'hello', artworkUrl100: 'artWork', trackId: 1 });
+  });
+
+  it('sshould call play audio on Pause Button click', () => {
```
Comment (rajatdangat-ws) [inline app/components/Track/tests/index.test.js]: fix typo

File: app/components/Track/tests/index.test.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    const element = renderWithIntl(<Track track={{ trackName: 'hello', artworkUrl100: 'artWork', trackId: 1 }} currentTrackID={1} />);
+    const pauseButtonEle = element.getAllByLabelText(`${translate('pauseText')} ${translate('buttonText')}`);
+    console.log(pauseButtonEle.length);
```
Comment (rajatdangat-ws) [inline app/components/Track/tests/index.test.js]: remove log if not required

File: app/translations/en.json [hunk 8]
```diff
@@ -18,5 +18,9 @@
   "default_template": "wednesday-solutions/react-template",
   "search_tunes": "Search Tunes",
   "search_song_detail": "Got a song stuck in your mind? Search Here:",
-  "tunes_search_placeholder": "Type song, artist, album, genre, etc."
+  "tunes_search_placeholder": "Type song, artist, album, genre, etc.",
+  "playText": "Play",
```
Comment (rajatdangat-ws) [inline app/translations/en.json]: please follow same format for the key names

**Pull request merged**